src/service/standard-actions-impl.js

```diff
@@ -88,14 +103,62 @@ export class StandardActions {
    */
   handleHide(invocation) {
     const target = dev().assertElement(invocation.target);
+
     this.resources_.mutateElement(target, () => {
       if (target.classList.contains('i-amphtml-element')) {
         target./*OK*/collapse();
       } else {
         toggle(target, false);
+        target.setAttribute('hidden', '');
       }
     });
   }
+
+  /**
+   * Handles "show" action. This is a very simple action where "display: none"
+   * is removed from the target element.
+   * @param {!./action-impl.ActionInvocation} invocation
+   */
+  handleShow(invocation) {
+    const target = dev().assertElement(invocation.target);
+
+    if (getMode().development) {
+      dev().assert(
+          !target.classList.contains(getLayoutClass(Layout.NODISPLAY)),
+          'Elements with layout=nodisplay cannot be dynamically shown. %s',
+          target);
+
+      dev().assert(
+          target.style.display == 'none' || target.hasAttribute('hidden'),
+          'Elements can only be dynamically shown when they have the "hidden"' +
+          ' attribute set or when they were previously dynamically hidden. %s',
+          target);
+    }
+
```

Comment:
So the case still not covered for the warning (hence my comment regarding computedStyle) is if element is hidden using a css class rather than inline-style, `target.style.display` will be empty. So doing a similar check to above using computedStyle in a vsync seems needed for completeness.